rascal-vscode-extension/src/RascalExtension.ts

```diff
@@ -99,6 +101,30 @@ export class RascalExtension implements vscode.Disposable {
         );
     }
 
+
+    private registerCopyLocation() {
+        this.context.subscriptions.push(
+            vscode.commands.registerTextEditorCommand("rascalmpl.copyLocation", (text, _edit, moduleName) => {
+                const editor = vscode.window.activeTextEditor;
+                if (editor) {
+                    const fileName = editor.document.uri.fsPath;
+                    const selection = editor.selection;
+
+                    let offset = editor.document.offsetAt(selection.start);
+                    let length = editor.document.getText(selection).length;
+                    let startLine = selection.start.line;
+                    let startColumn = selection.start.character;
+                    let endLine = selection.end.line;
+                    let endColumn = selection.end.character;
```

Comment:
This section code is sensitive to a difference between vscode & rascal. VS Code thinks in utf16 encoding, while rascal in utf32. In most cases, they are the same, until you encounter stuff like `" 🍝🚀 "` or other higher plane characters. 

In that case the offset, length, and columns are off by 1 or more. Since 🚀 in utf16 is 2 chars wide (the surrogate pair `0xD83D, 0xDE80`), while in utf32 its 1 char wide (`0x0001F680`). 

The pattern to fix this is:

1. fix `offset` by counting all utf16 surrogate pairs before this point, and increment offset by that count
2. fix `length` by counting the utf16 surrogate pairs inside the selection, and adding that to the length
3. fix `startColumn` by counting utf16 surrogate pairs before the selection *on `startLine`*,
4. fix `endColumn` by counting utf16 surrogate pairs before the end of the selection *on `endLine`*.


And good news, we've done this before, if you look at the functions in `RascalTerminalLinkProvider`, they do exactly this. Maybe part of this PR is that we refactor that to a separat module? and then use that here?